pkg/gui/custom_commands.go

```diff
@@ -151,10 +156,85 @@ func (gui *Gui) handleCustomCommandKeybinding(customCommand config.CustomCommand
 						return gui.surfaceError(err)
 					}
 
+					return gui.createMenu(title, menuItems, createMenuOptions{showCancel: true})
+				}
+			case "menuFromCommand":
+				f = func() error {
+					// Collect cmd to run from config
+					cmdStr, err := gui.resolveTemplate(prompt.Command, promptResponses)
+					if err != nil {
+						return gui.surfaceError(err)
+					}
+
+					// Collect Filter regexp
+					filter, err := gui.resolveTemplate(prompt.Filter, promptResponses)
+					if err != nil {
+						return gui.surfaceError(err)
+					}
+					reg, err := regexp.Compile(filter)
+					if err != nil {
+						return gui.surfaceError(errors.New("unable to parse filter regex, error: " + err.Error()))
+					}
+
+					// Run and save output
+					message, err := gui.GitCommand.RunCommandWithOutput(cmdStr)
+					if err != nil {
+						return gui.surfaceError(err)
+					}
+
+					// Need to make a menu out of what the cmd has displayed
```

Comment:
could we extract this part out and add a test i.e. a function that takes a command output and a format and returns a slice of candidate strings?

Reply:
See `generateMenuCandidates` function